modules/common/adapters/adapter.h

```diff
@@ -239,7 +240,21 @@ class Adapter : public AdapterBase {
         << ":" << topic_name_;
     return *observed_queue_.front();
   }
-
+  /**
+   * @brief returns the most recent message pointer in the observing queue.
+   *
+   * /note
+   * Please call Empty() to make sure that there is data in the
+   * queue before calling GetLatestObservedPtr().
+   */
+  std::shared_ptr<D> GetLatestObservedPtr() const {
```

Comment:
We'd better stick to using the const reference. Keep the "observed data" read-only.